airbyte-webapp/src/core/form/schemaToYup.test.ts

```diff
@@ -0,0 +1,269 @@
+            },
+          },
+        },
+      ],
+    },
+  },
+};
+
+it("should build  mixed schema for conditional case", () => {
+  const yupSchema = buildYupFormForJsonSchema(simpleConditionalSchema, jsonSchemaToFormBlock(simpleConditionalSchema));
+
+  const expectedSchema = yup.object().shape({
+    start_date: yup.string().trim().required("form.empty.error").transform(String),
+    max_objects: yup.number().transform((x) => x),
+    credentials: yup.object().shape({
+      type: yup.mixed(),
+      api_key: yup
+        .mixed()
+        .when("type", { is: "", then: (x) => x, otherwise: (x) => x })
+        .when("type", { is: "", then: (x) => x, otherwise: (x) => x }),
+      redirect_uri: yup
+        .mixed()
+        .when("type", { is: "", then: (x) => x, otherwise: (x) => x })
+        .when("type", { is: "", then: (x) => x, otherwise: (x) => x }),
+    }),
+  });
+
+  expect(JSON.parse(JSON.stringify(yupSchema))).toEqual(JSON.parse(JSON.stringify(expectedSchema)));
+});
+
+describe("should schema build conditional case that validates correctly based on selection key", () => {
+  const yupSchema = buildYupFormForJsonSchema(simpleConditionalSchema, jsonSchemaToFormBlock(simpleConditionalSchema));
+  it("enforces required props for selected condition", () => {
+    expect(() => {
+      yupSchema.validateSync({
+        start_date: "2022",
+        max_objects: 5,
+        credentials: {
+          // api needs api_key, so this should fail
+          type: "api",
+          redirect_uri: "test",
+        },
+      });
+    }).toThrowError("form.empty.error");
+  });
+
+  it("does not enforce additional contraints if the condition is selected", () => {
+    expect(() => {
+      yupSchema.validateSync({
+        start_date: "2022",
+        max_objects: 5,
+        credentials: {
+          type: "oauth",
+          redirect_uri: "test",
+          // does not match the pattern, but it should not be validated
+          api_key: "X",
+        },
+      });
+    }).not.toThrowError();
+  });
+
+  it("enforces additional contraints only if the condition is selected", () => {
+    expect(() => {
+      yupSchema.validateSync({
+        start_date: "2022",
+        max_objects: 5,
+        credentials: {
+          type: "api",
+          // does not match the pattern, so it should fail
+          api_key: "X",
+        },
+      });
+    }).toThrowError("form.pattern.error");
+  });
+
+  it("strips out properties belonging to other conditions", () => {
+    const cleanedValues = yupSchema.cast(
+      {
+        start_date: "2022",
+        max_objects: 5,
+        credentials: {
+          type: "api",
+          api_key: "X",
+          redirect_uri: "test",
+        },
+      },
+      {
+        stripUnknown: true,
+      }
+    );
+    expect(cleanedValues.credentials).toEqual({
+      type: "api",
+      api_key: "X",
+    });
+  });
+});
+
+it("should build schema for conditional case with inner schema and form blocks", () => {
```

Comment:
This test implementation doesn't seem to line up with this description to me. I'm not sure what `topKey` and `topKey.subKey` are doing on lines 250-251. It seems like this should be nesting `credentials` under a few levels or something?

Reply:
I copied this test over and didn't adjust much about it. It is testing whether conditionals also work in nested cases, but I don't see how we need that test as the base schema is also nesting the oneOf in an object...

Removed the test as it's more confusing than helpful.